Bug when minimizing in rewriter?

https://github.com/coq/coq/pull/17346#issuecomment-1533755349 has Qed lemmas left

I'm not sure this is a bug in the minimizer?  It seems just as likely that it's a bug in the runner infrastructure or in Coq itself, or perhaps in the way the rewriter is set up. The actual issue is that when minimization resumed, when it ran
```
/github/workspace/builds/coq/coq-passing/_install_ci/bin/coq_makefile COQC = /github/workspace/builds/coq/coq-passing/_install_ci/bin/coqc.orig COQDEP = /github/workspace/builds/coq/coq-passing/_install_ci/bin/coqdep -o Makefile44j2imgx.coq -R /github/workspace/builds/coq/coq-passing/_build_ci/rewriter/src/Rewriter Rewriter -Q /github/workspace/cwd Top -Q /github/workspace/builds/coq/coq-passing/_install_ci/lib/coq/user-contrib/Ltac2 Ltac2 -Q /github/workspace/builds/coq/coq-passing/_install_ci/lib/coq/user-contrib/Rewriter Rewriter -I /github/workspace/builds/coq/coq-passing/_build_ci/rewriter/src/Rewriter/Util/plugins -R /github/workspace/builds/coq/coq-passing/_build_ci/rewriter/src/Rewriter Rewriter -Q /github/workspace/cwd Top -Q /github/workspace/builds/coq/coq-passing/_install_ci/lib/coq/user-contrib/Ltac2 Ltac2 -Q /github/workspace/builds/coq/coq-passing/_install_ci/lib/coq/user-contrib/Rewriter Rewriter -I /github/workspace/builds/coq/coq-passing/_build_ci/rewriter/src/Rewriter/Util/plugins -arg -q -arg -w -arg +implicit-core-hint-db,+implicits-in-term,+non-reversible-notation,+deprecated-intros-until-0,+deprecated-focus,+unused-intro-pattern,+deprecated-hint-constr,+fragile-hint-constr,+variable-collision,+unexpected-implicit-declaration,+omega-is-deprecated,+deprecated-instantiate-syntax,+non-recursive,+deprecated-hint-rewrite-without-locality,+deprecated-hint-without-locality,+deprecated-instance-without-locality,+undeclared-scope,+deprecated-typeclasses-transparency-without-locality,+future-coercion-class-field,unsupported-attributes -arg -w -arg -notation-overridden,-unusable-identifier -arg -w -arg -notation-overridden -arg -w -arg -deprecated-native-compiler-option -arg -native-compiler -arg ondemand bug_01.v
```
followed by
```
make -k -f Makefile44j2imgx.coq KEEP_ERROR=1 bug_01.glob
```
the result was (after many "required file matches multiple file in path" warnings)
```
*** Error: 
in file bug_01.v, could not find META.coq-rewriter.
make: *** No rule to make target 'bug_01.glob'.
```
and as a result the minimizer was unable to understand the `Require`s.  So because `coqdep` fails fatally with this error, `make` never gets around to making the `.glob` file.

--- a/find-bug.py
+++ b/find-bug.py
@@ -116,7 +116,7 @@ parser.add_argument('--strict-whitespace', dest='strict_whitespace',
                           "messages.  By default, locations where there " +
                           "are newlines followed by spaces are interchangable " +
                           "with any amount of spacing."))
-parser.add_argument('--no-deps', dest='walk_tree',
+parser.add_argument('--no-deps', dest='walk_tree_and_use_coq_makefile',
                     action='store_const', const=False, default=True,
                     help=("Don't do dependency analysis on all files in the current " +
                           "file tree."))
@@ -1327,7 +1327,7 @@ if __name__ == '__main__':
         'base_dir': (os.path.abspath(args.base_dir)
                      if args.base_dir != ''
                      else None),
-        'walk_tree': args.walk_tree,
+        'walk_tree_and_use_coq_makefile': args.walk_tree_and_use_coq_makefile,
         'strict_whitespace': args.strict_whitespace,
         'temp_file_name': args.temp_file,
         'coqc_is_coqtop': args.coqc_is_coqtop,

--- a/find-bug.py
+++ b/find-bug.py
@@ -116,10 +116,12 @@ parser.add_argument('--strict-whitespace', dest='strict_whitespace',
                           "messages.  By default, locations where there " +
                           "are newlines followed by spaces are interchangable " +
                           "with any amount of spacing."))
-parser.add_argument('--no-deps', dest='walk_tree_and_use_coq_makefile',
+parser.add_argument('--no-deps', dest='use_coq_makefile_for_deps',
                     action='store_const', const=False, default=True,
-                    help=("Don't do dependency analysis on all files in the current " +
-                          "file tree."))
+                    help=("Don't do dependency analysis with coq_makefile."))
+parser.add_argument('--no-pwd-deps', dest='walk_tree',
+                    action='store_const', const=False, default=True,
+                    help=("Don't add all files in the current directory to the dependency analysis."))
 parser.add_argument('--inline-coqlib', dest='inline_coqlib',
                     action='store_const', const=True, default=False,
                     help=("Attempt to inline requires from Coq's standard library"))
@@ -1327,7 +1329,8 @@ if __name__ == '__main__':
         'base_dir': (os.path.abspath(args.base_dir)
                      if args.base_dir != ''
                      else None),
-        'walk_tree_and_use_coq_makefile': args.walk_tree_and_use_coq_makefile,
+        'use_coq_makefile_for_deps': args.use_coq_makefile_for_deps,
+        'walk_tree': args.walk_tree,
         'strict_whitespace': args.strict_whitespace,
         'temp_file_name': args.temp_file,
         'coqc_is_coqtop': args.coqc_is_coqtop,

--- a/import_util.py
+++ b/import_util.py
@@ -33,15 +33,16 @@ def error(*objs):
 
 def fill_kwargs(kwargs, for_makefile=False):
     rtn = {
-        'libnames'                       : DEFAULT_LIBNAMES,
-        'non_recursive_libnames'         : tuple(),
-        'ocaml_dirnames'                 : tuple(),
-        'log'                            : DEFAULT_LOG,
-        'coqc'                           : 'coqc',
-        'coq_makefile'                   : 'coq_makefile',
-        'coqdep'                         : 'coqdep',
-        'walk_tree_and_use_coq_makefile' : True,
-        'coqc_args'                      : tuple(),
+        'libnames'                  : DEFAULT_LIBNAMES,
+        'non_recursive_libnames'    : tuple(),
+        'ocaml_dirnames'            : tuple(),
+        'log'                       : DEFAULT_LOG,
+        'coqc'                      : 'coqc',
+        'coq_makefile'              : 'coq_makefile',
+        'coqdep'                    : 'coqdep',
+        'use_coq_makefile_for_deps' : True,
+        'walk_tree'                 : True,
+        'coqc_args'                 : tuple(),
     }
     rtn.update(kwargs)
     if for_makefile:
@@ -471,10 +472,10 @@ def make_globs(logical_names, **kwargs):
     filenames_v    = [v_name    for vo_name, v_name, glob_name in filenames_vo_v_glob]
     filenames_glob = [glob_name for vo_name, v_name, glob_name in filenames_vo_v_glob]
     if len(filenames_vo_v_glob) == 0: return
-    extra_filenames_v = list(get_all_v_files('.', filenames_v) if kwargs['walk_tree_and_use_coq_makefile'] else [])
+    extra_filenames_v = list(get_all_v_files('.', filenames_v) if kwargs['use_coq_makefile_for_deps'] and kwargs['walk_tree'] else [])
     all_filenames_v = filenames_v + extra_filenames_v
 
-    if len(all_filenames_v) <= 1 or not kwargs['walk_tree_and_use_coq_makefile']:
+    if len(all_filenames_v) <= 1 or not kwargs['use_coq_makefile_for_deps']:
         if len(all_filenames_v) > 1:
             kwargs['log']("WARNING: Making globs for %s separately without coq_makefile for dependencies, as requested" % ", ".join(all_filenames_v))
         for v_name in all_filenames_v:

--- a/import_util.py
+++ b/import_util.py
@@ -33,15 +33,15 @@ def error(*objs):
 
 def fill_kwargs(kwargs, for_makefile=False):
     rtn = {
-        'libnames'              : DEFAULT_LIBNAMES,
-        'non_recursive_libnames': tuple(),
-        'ocaml_dirnames'        : tuple(),
-        'log'                   : DEFAULT_LOG,
-        'coqc'                  : 'coqc',
-        'coq_makefile'          : 'coq_makefile',
-        'coqdep'                : 'coqdep',
-        'walk_tree'             : True,
-        'coqc_args'             : tuple(),
+        'libnames'                       : DEFAULT_LIBNAMES,
+        'non_recursive_libnames'         : tuple(),
+        'ocaml_dirnames'                 : tuple(),
+        'log'                            : DEFAULT_LOG,
+        'coqc'                           : 'coqc',
+        'coq_makefile'                   : 'coq_makefile',
+        'coqdep'                         : 'coqdep',
+        'walk_tree_and_use_coq_makefile' : True,
+        'coqc_args'                      : tuple(),
     }
     rtn.update(kwargs)
     if for_makefile:
@@ -471,8 +471,16 @@ def make_globs(logical_names, **kwargs):
     filenames_v    = [v_name    for vo_name, v_name, glob_name in filenames_vo_v_glob]
     filenames_glob = [glob_name for vo_name, v_name, glob_name in filenames_vo_v_glob]
     if len(filenames_vo_v_glob) == 0: return
-    extra_filenames_v = (get_all_v_files('.', filenames_v) if kwargs['walk_tree'] else [])
-    (stdout_make, stderr_make) = run_coq_makefile_and_make(tuple(sorted(list(filenames_v) + list(extra_filenames_v))), filenames_glob, **kwargs)
+    extra_filenames_v = list(get_all_v_files('.', filenames_v) if kwargs['walk_tree_and_use_coq_makefile'] else [])
+    all_filenames_v = filenames_v + extra_filenames_v
+
+    if len(all_filenames_v) <= 1 or not kwargs['walk_tree_and_use_coq_makefile']:
+        if len(all_filenames_v) > 1:
+            kwargs['log']("WARNING: Making globs for %s separately without coq_makefile for dependencies, as requested" % ", ".join(all_filenames_v))
+        for v_name in all_filenames_v:
+            make_one_glob_file(v_name, **kwargs)
+    else:
+        (stdout_make, stderr_make) = run_coq_makefile_and_make(tuple(sorted(filenames_v + extra_filenames_v)), filenames_glob, **kwargs)
 
 def get_glob_file_for(filename, update_globs=False, **kwargs):
     kwargs = fill_kwargs(kwargs)

--- a/inline-imports.py
+++ b/inline-imports.py
@@ -20,10 +20,12 @@ parser.add_argument('output_file', metavar='OUT_FILE', type=argparse.FileType('w
 parser.add_argument('--fast-merge-imports', dest='fast_merge_imports',
                     action='store_const', const=True, default=False,
                     help='Use a faster method for combining imports')
-parser.add_argument('--no-deps', dest='walk_tree_and_use_coq_makefile',
+parser.add_argument('--no-deps', dest='use_coq_makefile_for_deps',
                     action='store_const', const=False, default=True,
-                    help=("Don't do dependency analysis on all files in the current " +
-                          "file tree."))
+                    help=("Don't do dependency analysis with coq_makefile."))
+parser.add_argument('--no-pwd-deps', dest='walk_tree',
+                    action='store_const', const=False, default=True,
+                    help=("Don't add all files in the current directory to the dependency analysis."))
 parser.add_argument('--no-wrap-modules', dest='wrap_modules',
                     action='store_const', const=False, default=True,
                     help=("Don't wrap imports in Modules.  By default, the " +
@@ -66,7 +68,8 @@ if __name__ == '__main__':
         'as_modules': args.wrap_modules,
         'fast': args.fast_merge_imports,
         'coq_makefile': args.coq_makefile,
-        'walk_tree_and_use_coq_makefile': args.walk_tree_and_use_coq_makefile
+        'use_coq_makefile_for_deps': args.use_coq_makefile_for_deps,
+        'walk_tree': args.walk_tree,
         }
     update_env_with_libnames(env, args)
     if args.inline_user_contrib: update_env_with_coqpath_folders(env, os.path.join(get_coqc_coqlib(env['coqc'], coq_args=env['coqc_args'], **env), 'user-contrib'))

--- a/inline-imports.py
+++ b/inline-imports.py
@@ -20,7 +20,7 @@ parser.add_argument('output_file', metavar='OUT_FILE', type=argparse.FileType('w
 parser.add_argument('--fast-merge-imports', dest='fast_merge_imports',
                     action='store_const', const=True, default=False,
                     help='Use a faster method for combining imports')
-parser.add_argument('--no-deps', dest='walk_tree',
+parser.add_argument('--no-deps', dest='walk_tree_and_use_coq_makefile',
                     action='store_const', const=False, default=True,
                     help=("Don't do dependency analysis on all files in the current " +
                           "file tree."))
@@ -66,7 +66,7 @@ if __name__ == '__main__':
         'as_modules': args.wrap_modules,
         'fast': args.fast_merge_imports,
         'coq_makefile': args.coq_makefile,
-        'walk_tree': args.walk_tree
+        'walk_tree_and_use_coq_makefile': args.walk_tree_and_use_coq_makefile
         }
     update_env_with_libnames(env, args)
     if args.inline_user_contrib: update_env_with_coqpath_folders(env, os.path.join(get_coqc_coqlib(env['coqc'], coq_args=env['coqc_args'], **env), 'user-contrib'))

